src/utils/helpers/DetailsDrawHelpers.ts

```diff
@@ -0,0 +1,223 @@
+import { TreeNode } from '../../components/types';
+import { selectDetailsContainerDiv, selectDetailsExitButtonWrapper, selectDetailsViewContainer, selectDetailsZoom } from './Selectors';
+import { ElementColors, FAST_TRANSITION_DURATION, Selectors } from '../AppConsts';
+import { hierarchy, HierarchyPointNode, tree } from 'd3-hierarchy';
+import { create, linkHorizontal, zoom, zoomIdentity, symbol, symbolCross } from 'd3';
+import { createZoom } from './DrawHelpers';
+import { changeZoom } from './GraphHelpers';
+
+export function createDetailsViewContainer(width: number, height: number) {
+    const containerWidth = width * 4;
+    const containerHeight = height * 4;
+    const container = selectDetailsContainerDiv()
+        .style('display', 'none')
+        .style('opacity', 0)
+        .append('svg')
+        .attr('id', 'details-view-container')
+        .attr('x', 0)
+        .attr('y', 0)
+        .attr('width', containerWidth)
+        .attr('height', containerHeight)
+        .attr('viewBox', `0 0 ${width} ${height}`)
+        .attr('preserveAspectRatio', 'xMidYMid meet');
+    container
+        .append('rect')
+        .attr('width', width)
+        .attr('height', height)
+        .attr('fill', ElementColors.DETAILS_BACKGROUND);
+    container
+        .append('g')
+        .attr('id', 'details-exit-button')
+        .attr('cursor', 'pointer')
+        .append('rect')
+        .attr('transform', 'translate(5,5)')
+        .attr('width', 10)
+        .attr('height', 10)
+        .attr('fill', ElementColors.DETAILS_BACKGROUND);
+    selectDetailsExitButtonWrapper()
+        .append('path')
+        .attr('transform', 'translate(10,10) rotate(45)')
+        .attr(
+            'd',
+            symbol()
+                .type(symbolCross)
+                .size(20)
+        );
+    createZoom(container, Selectors.ZOOM_DETAILS);
+    container.call(
+        zoom<any, any>().on('zoom', changeZoom(Selectors.ZOOM_DETAILS)).transform,
+        // rough center screen on diagram's root node
+        zoomIdentity.translate(-width / 5.3, -height / 2.65)
+    );
+}
+
+interface TreeStructure {
+    name: string;
+    children?: TreeStructure[];
+}
+
+type TreeNodeWithVisitedFlag = TreeNode & { isVisited?: boolean };
+
+function mapNodeToTreeStructure(node: TreeNodeWithVisitedFlag, links: (keyof TreeNode & 'consumers') | 'providers'): TreeStructure {
```

Comment:
links can't be just 'consumers' | 'providers ?

Reply:
yes, they can - will change it